ddtrace/tracer.py

```diff
@@ -761,7 +761,13 @@ def set_tags(self, tags):
     def shutdown(self, timeout=None):
         """Shutdown the tracer.
 
-        This will stop the background writer/worker and flush any finished traces in the buffer.
+        This will stop the background writer/worker and flush any finished
+        traces in the buffer.
+
+        Note that the tracer *cannot* be used after it has been shutdown. Any
+        traces generated after a shutdown will result in a ``RuntimeError``
+        being raised on ``span.finish()`` (or when the span context manager
+        exits).
```

Comment:
This is a little bit too specific to the current writer. We should take a step back here and see what's the `Writer` API would be. So far we started with `write` and `stop`, which is fine. Then, your *base* class should handle that assertion of "write() can never be called after stop()".
Remember that a base class, even abstract, can be called with `super()`. So you could have

```python
@attr.s
class Writer(abc.ABC):
     started = attr.ib(type=bool, default=False)

     @abc.abastractmethod
     def write(self):
         if not self.started:
              raise TracerStopped
```

I'd suggest using the base `Service` class from the profiler that handles a `status` attribute that is a useful base.

Then, the error raised should be more specific than just `RuntimeError`. I'd start with just e.g.:

```python
class TracerStopped(RuntimeError):
     pass
```